wiki/src/components/Twelf.astro

```diff
@@ -10,14 +10,58 @@ function trimSlightly(code: string): string {
   return code3;
 }
 
-const { code = "" } = Astro.props;
+const { context = "", code = "" } = Astro.props;
 function htmlEscape(code: string) {
   return code
     .replaceAll("&", "&amp;")
     .replaceAll("<", "&lt;")
     .replaceAll(">", "&gt;");
 }
 
+function bytesOfString(str: string): Uint8Array {
+  return new TextEncoder().encode(str);
+}
+
+function concatenateUint8Arrays(arrays: Uint8Array[]) {
+  if (arrays.length == 0) return new Uint8Array([]);
+  const totalLength = arrays.map((arr) => arr.length).reduce((a, b) => a + b);
+  const result = new Uint8Array(totalLength);
+  let offset = 0;
+  for (const arr of arrays) {
+    result.set(arr, offset);
+    offset += arr.length;
+  }
+  return result;
+}
+
+async function bytesOfStream(
+  readableStream: ReadableStream
+): Promise<Uint8Array> {
+  const reader = readableStream.getReader();
+  let result: Uint8Array[] = [];
+  while (true) {
+    const { done, value } = await reader.read();
+    if (done) break;
+    result.push(value);
+  }
+  return concatenateUint8Arrays(result);
+}
```

Comment:
Copied from https://github.com/jcreedcmu/twelf-wasm/blob/main/src/encoding.ts#L29

Reply:
I might suggest putting that link in a comment in the code itself to save an archaeology step.

I note that we are consciously rejecting, for the time being, the alternative approaches of:
1) *including* like all of twelf-wasm as a submodule
2) copying and pasting the whole `encoding.ts` here

I almost thought about recommending (2) but eh, we're only encoding not encoding-and-decoding, so I don't really have a problem with the narrower scope copy-paste.

Reply:
refactoring and adding a note is absolutely the right call here